python/ray/serve/tests/test_deployment_graph_autoscaling.py

```diff
@@ -0,0 +1,227 @@
+import sys
+import pytest
+
+from python.ray.serve import constants as serve_constants
+
+import ray
+from ray import serve
+from ray.serve.drivers import DAGDriver
+from ray.experimental.dag.input_node import InputNode
+from ray.serve.deployment_state import ReplicaState
+from ray._private.test_utils import SignalActor, wait_for_condition
+
+
+def get_num_running_replicas(controller, deployment_name):
+    replicas = ray.get(
+        controller._dump_replica_states_for_testing.remote(deployment_name)
+    )
+    running_replicas = replicas.get([ReplicaState.RUNNING])
+    return len(running_replicas)
+
+
+def test_autoscaling_0_replica(serve_instance):
+    serve_constants.HANDLE_METRIC_PUSH_INTERVAL_S = 1
```

Comment:
Nit: Could this affect other tests?  Do we need to use pytest `monkeypatch` fixture or something similar?

Reply:
Also could you add a comment saying why it's important to set this to 1 here?

Reply:
deployment graph is a little special, potentially we can have long chain nodes, if keep HANDLE_METRIC_PUSH_INTERVAL_S, scaling from 0 will take long time for each node.

Magic number I just choose to work.

I will leave some comments in the code